internal/app/machined/pkg/controllers/network/hostname_config.go

```diff
@@ -172,6 +209,21 @@ func (ctrl *HostnameConfigController) apply(ctx context.Context, r controller.Ru
 	return ids, nil
 }
 
+func (ctrl *HostnameConfigController) getStableDefault(machineID []byte) (*network.HostnameSpecSpec, error) {
+	id := string(machineID)
+
+	if len(id) < 6 {
+		return nil, fmt.Errorf("machine ID is too short: %s", id)
+	}
+
+	hostname := "talos-" + id[0:6]
```

Comment:
machineID is 0-9a-f, I think too little values for our purpose.

I would propose to do `talos-XXX-XXX` where each `X` is `a-z0-9`, this gives us 36^6 different hostnames. 

in current scheme we have 16^6, which is way smaller.

Reply:
Got it, so I guess first 6 chars of `lower(base36(nodeID))` with `-` in the middle.

Do you think hashing (sha256 or something) the nodeID beforehand  make sense? To avoid any kind of potential length issues etc.

Reply:
we can do hashing to prevent a leak of node identity via hostname (which might be sent out to the DHCP server)

Reply:
Yep, I thought so 👍